### PR TITLE
Release 5.4.0

### PR DESCRIPTION
**Pull request by mgreisen**
## Summary of the Pull Request

Version and change log updates for release 5.4.0.

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed
check-pr

_How does someone test & validate?_

### CODE REVIEW

**Comment by chkeita** [comment]
[this PR](https://github.com/microsoft/onefuzz/pull/1912/files) should be part of this release.
cc: @tevoinea

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,25 @@ All notable changes to this project will be documented in this file.
...
+* Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#1619](https://github.com/microsoft/onefuzz/pull/1619), [#1644](https://github.com/microsoft/onefuzz/pull/1644), [#1645](https://github.com/microsoft/onefuzz/pull/1645), [#1646](https://github.com/microsoft/onefuzz/pull/1646), [#1655](https://github.com/microsoft/onefuzz/pull/1655), [#1700](https://github.com/microsoft/onefuzz/pull/1700), [#1849](https://github.com/microsoft/onefuzz/pull/1849), [#1882](https://github.com/microsoft/onefuzz/pull/1882)
+* Agent: Separate `llvm-symbolizer` setup from sanitizer environment variable initialization. [#1778](https://github.com/microsoft/onefuzz/pull/1778)
+* Agent: Set the TSan options based on the external symbolizer. [#1787](https://github.com/microsoft/onefuzz/pull/1787)
+* Service: Removed the `client_secret` from the configuration file to eliminate the potential for the secret to be persisted. If you have persisted a client secret in your configuration file, remove it. [#1918](https://github.com/microsoft/onefuzz/pull/1918)
+* Service: Use a sas url to download log files. [#1920](https://github.com/microsoft/onefuzz/pull/1920)
```
Comment (tevoinea) [inline CHANGELOG.md:20]:
```suggestion
* CLI: Use a sas url to download log files. [#1920](https://github.com/microsoft/onefuzz/pull/1920)
```

**Review by tevoinea — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,25 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 5.4.0
+### Added
+* Agent: Added the OneFuzz version and tool name to the Crash Report. [#1635](https://github.com/microsoft/onefuzz/pull/1635)
+* Agent: Added a check for missing libraries when running the `-help` check. [#1812](https://github.com/microsoft/onefuzz/pull/1812)
```
Comment (ranweiler) [inline CHANGELOG.md:10]:
```suggestion
* Agent: Added a check for missing libraries when running the LibFuzzer `-help` check. [#1812](https://github.com/microsoft/onefuzz/pull/1812)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,25 @@ All notable changes to this project will be documented in this file.
...
+* CLI/Service: Updated multiple first-party and third-party Python dependencies. [#1784](https://github.com/microsoft/onefuzz/pull/1784)
+* Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#1619](https://github.com/microsoft/onefuzz/pull/1619), [#1644](https://github.com/microsoft/onefuzz/pull/1644), [#1645](https://github.com/microsoft/onefuzz/pull/1645), [#1646](https://github.com/microsoft/onefuzz/pull/1646), [#1655](https://github.com/microsoft/onefuzz/pull/1655), [#1700](https://github.com/microsoft/onefuzz/pull/1700), [#1849](https://github.com/microsoft/onefuzz/pull/1849), [#1882](https://github.com/microsoft/onefuzz/pull/1882)
+* Agent: Separate `llvm-symbolizer` setup from sanitizer environment variable initialization. [#1778](https://github.com/microsoft/onefuzz/pull/1778)
+* Agent: Set the TSan options based on the external symbolizer. [#1787](https://github.com/microsoft/onefuzz/pull/1787)
+* Service: Removed the `client_secret` from the configuration file to eliminate the potential for the secret to be persisted. If you have persisted a client secret in your configuration file, remove it. [#1918](https://github.com/microsoft/onefuzz/pull/1918)
```
Comment (ranweiler) [inline CHANGELOG.md:19]:
```suggestion
* CLI: Added the `ONEFUZZ_CLIENT_SECRET` environment variable and removed the `client_secret` field from the configuration file. This prevents accidental misuse via persisting the secret to disk outside of [confidential client](https://docs.microsoft.com/en-us/azure/active-directory/develop/developer-glossary#client-application) environments. If you have set a client secret in your configuration file in a public client, we recommending removing and revoking it. CI scripts that currently set the client secret in the config must instead pass it via the `ONEFUZZ_CLIENT_SECRET` environment variable or on each CLI invocation via the `--client_secret` argument. [#1918](https://github.com/microsoft/onefuzz/pull/1918)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,25 @@ All notable changes to this project will be documented in this file.
...
+* Service: Use a sas url to download log files. [#1920](https://github.com/microsoft/onefuzz/pull/1920)
+### Fixed
+* Agent: Only watch directories for change events. [#1859](https://github.com/microsoft/onefuzz/pull/1859)
+* Agent: Switch to a smart constructor to minimize miss-use. [#1865](https://github.com/microsoft/onefuzz/pull/1865)
```
Comment (ranweiler) [inline CHANGELOG.md:23]:
```suggestion
* Agent: Switch to a smart constructor to minimize misuse. [#1865](https://github.com/microsoft/onefuzz/pull/1865)
```

File: CHANGELOG.md (hunk 1 above)
Comment (ranweiler) [reply CHANGELOG.md:20]:
```suggestion
* CLI: Use a SAS URL to download log files. [#1920](https://github.com/microsoft/onefuzz/pull/1920)
```
(apply to any other use of acronyms like "SAS", "URL", "CLI")

**Review by chkeita — APPROVED** [review]

**Review by ranweiler — APPROVED** [review]

**Pull request merged**